### PR TITLE
Create standalone class for CallConv parsing.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
This reworks the API for CallConv into a standalone class and removes the macro expansion approach for constructing the logic.

/cc @jkoritzinsky @elinor-fung

### CODE REVIEW

File: src/coreclr/vm/siginfo.hpp [hunk 1]
```diff
@@ -1208,6 +1184,51 @@ class MetaSig
...
+        CALL_CONV_MOD_NONE = 0,
+        CALL_CONV_MOD_SUPPRESSGCTRANSITION = 0x1,
+        CALL_CONV_MOD_MEMBERFUNCTION = 0x2
+    };
+
+    struct State
+    {
+        CorInfoCallConvExtension CallConvBase;
+        CallConvModifiers CallConvModifiers;
+    };
```
Comment (jkoritzinsky) [inline src/coreclr/vm/siginfo.hpp:1206]: The `State` struct should be able to be private to `CallConvBuilder`.

**Review by jkoritzinsky — APPROVED** [review]
LGTM with one suggestion.

File: src/coreclr/vm/siginfo.hpp (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/siginfo.hpp:1206]: Yes I could. but then the internal templated `ProcessName` function would need to be a friend function. Making the nested type public seemed like a reasonable compromise between that and making it a high level one.

Comment (jkoritzinsky) [reply src/coreclr/vm/siginfo.hpp:1206]: I agree, that's a good reason. We can merge this as is then.

**Pull request merged**